File: Compiler_PJ_3/benchmarks/bubble_sort.py
n = 20
arr = list(range(n, 0, -1))
 
for j in range(n):
    for i in range(n - 1):
        if arr[i] > arr[i + 1]:
            arr[i], arr[i + 1] = arr[i + 1], arr[i]
 
for x in arr:
    print(x)

File: Compiler_PJ_3/benchmarks/bubble_sort.c
#include <stdio.h>
 
int main() {
    int n = 20;
    int arr[20];
    for (int i = 0; i < n; i++)
        arr[i] = n - i;
 
    for (int j = 0; j < n; j++)
        for (int i = 0; i < n - 1; i++)
            if (arr[i] > arr[i + 1]) {
                int tmp = arr[i];
                arr[i] = arr[i + 1];
                arr[i + 1] = tmp;
            }
 
    for (int i = 0; i < n; i++)
        printf("%d\n", arr[i]);
    return 0;
}